utils/record/record.go

```diff
@@ -233,13 +238,77 @@ func (e *EventRecorderAdapter) K8sRecorder() record.EventRecorder {
 	return e.Recorder
 }
 
-func NewAPIFactorySettings() api.Settings {
+func getAnalysisRunsFilterWithLabels(ro v1alpha1.Rollout, arInformer argoinformers.AnalysisRunInformer) (interface{}, error) {
+
+	set := labels.Set(map[string]string{
+		RolloutsPodTemplateHash: ro.Status.CurrentPodHash,
+	})
+
+	ars, err := arInformer.Lister().AnalysisRuns(ro.Namespace).List(labels.SelectorFromSet(set))
+	if err != nil || len(ars) == 0 {
+		return nil, err
+	}
+
+	revision, _ := annotations.GetRevisionAnnotation(&ro)
+	filteredArs := make([]*v1alpha1.AnalysisRun, 0, len(ars))
+
+	for _, ar := range ars {
+		arRevision, _ := annotations.GetRevisionAnnotation(ar)
+		if arRevision == revision {
+			filteredArs = append(filteredArs, ar)
+		}
+	}
+
+	sort.Slice(filteredArs[:], func(i, j int) bool {
+		ts1 := filteredArs[i].ObjectMeta.CreationTimestamp.Time
+		ts2 := filteredArs[j].ObjectMeta.CreationTimestamp.Time
+		return ts1.After(ts2)
+	})
+
+	var arsObj interface{}
+	arBytes, _ := json.Marshal(filteredArs)
+
+	if err != nil {
```

Comment:
This error is not checked, can we return an error and add context to the error message